### PR TITLE
Added tests to verify MAX MTU packet handling

### PR DESCRIPTION
**Pull request by prsunny**
Test sends a jumbo ICMP echo frame with MAX MTU size and expects the reply back. It also sends a jumbo frame to a route destination for verifying the forwarding functionality.

For the device configured with IP-MTU=9100, PHY-MTU=9122,  
- ICMP frame, the packet-len is 9114 (Subtracting 8 bytes of ICMP header from 9122) 
- IP frame, the packet-len is 9122 (This includes the Layer 2 Ethernet header + FCS) 

Note: For the test case to run, the PTF option must be set for socket buffer size to 16384

### CODE REVIEW

File: ansible/roles/test/files/ptftests/mtu_test.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    def setUp(self):
+        self.dataplane = ptf.dataplane_instance
+        self.router_mac = self.test_params['router_mac']
+    
+    #---------------------------------------------------------------------
+
+    def check_icmp_mtu(self):
+        '''
+        @summary: Check ICMP/Ping works for MAX MTU.
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/mtu_test.py]: more verbose description.

Check ICMP/Ping DUT works for MAX MTU

File: ansible/roles/test/files/ptftests/mtu_test.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        masked_exp_pkt = Mask(exp_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "id")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.ICMP, "chksum")
+         
+        src_port = 0
+        send_packet(self, src_port, pkt)
+        logging.info("Sending packet from port " + str(src_port) + " to " + ip_dst)
+        dst_port_list = [0,1]
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/mtu_test.py]: why are there two ports in the list? the icmp reply packet should arrive from the src_port as well. It should not be received at other ports.

File: ansible/roles/test/files/ptftests/mtu_test.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                            ip_src=ip_dst,
+                            ip_dst=ip_src,
+                            ip_ttl=64,
+                            icmp_type=0)
+
+        masked_exp_pkt = Mask(exp_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "id")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.ICMP, "chksum")
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/mtu_test.py]: maybe we should match the ttl as well, it is not guaranteed that the icmp reply also use ttl=64. You do not want that factor to fail your test.

File: ansible/roles/test/files/ptftests/mtu_test.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        '''
+        @summary: Check ICMP/Ping works for MAX MTU.
+        '''
+        ip_src = "10.0.0.1"
+        ip_dst = "10.0.0.0"
+        src_mac = self.dataplane.get_mac(0, 0)
+
+        pktlen = (self.DEFAULT_PACKET_LEN - self.ICMP_HDR_LEN)
+
+        pkt = simple_icmp_packet(pktlen=pktlen,
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/mtu_test.py]: should we set the DF bit in the packet to make sure we are not fragment the packet anywhere and we were not really testing the MTU sized packet?

File: ansible/roles/test/tasks/mtu.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+- name: Expand ToR properties into props
+  set_fact: props_tor="{{configuration_properties['tor']}}"
+  when: testbed_type in ['t1', 't1-lag']
+
+- name: Gathering minigraph facts about the device
+  minigraph_facts: host={{ inventory_hostname }}
+  connection: local
+
+- debug : msg="Start MTU Test"
```
Comment (lguohan) [inline ansible/roles/test/tasks/mtu.yml]: can we first get the mtu Ethernet0 on the DUT use the mtu as a parameter to send the packet. Then, it makes our test more generic?

You can get the mtu via ansible_facts, check below.

https://stackoverflow.com/questions/18839509/where-can-i-get-a-list-of-ansible-pre-defined-variables

File: ansible/roles/test/files/ptftests/mtu_test.py (hunk 1 above)
Comment (prsunny) [reply ansible/roles/test/files/ptftests/mtu_test.py]: Sure. Will modify the description

File: ansible/roles/test/files/ptftests/mtu_test.py (hunk 2 above)
Comment (prsunny) [reply ansible/roles/test/files/ptftests/mtu_test.py]: In t1-lag topology, 0 and 1 are part of PortChannel0. The reply packet can be received on either port. In t1 topology, the ICMP reply will be received on 0

File: ansible/roles/test/files/ptftests/mtu_test.py (hunk 4 above)
Comment (prsunny) [reply ansible/roles/test/files/ptftests/mtu_test.py]: I don't see a param in the API to set the DF bit. Will discuss with you!

File: ansible/roles/test/files/ptftests/mtu_test.py (hunk 3 above)
Comment (prsunny) [reply ansible/roles/test/files/ptftests/mtu_test.py]: Yes. Will add "ttl" to _do_not_care field in the reply packet

File: ansible/roles/test/tasks/mtu.yml (hunk 5 above)
Comment (prsunny) [reply ansible/roles/test/tasks/mtu.yml]: This shall be addressed in the next phase commit

**Pull request merged**